python/pynessie/cli_common_context.py

```diff
@@ -77,3 +77,24 @@ def parse_args(self: "DefaultHelp", ctx: click.Context, args: List) -> List:
 
 
 pass_client = click.make_pass_decorator(ContextObject)
+
+
+def validate_options(f: Callable) -> Callable:
+    """This Decorators is responsible to validate CLI options and override them with default if needed."""
+
+    @functools.wraps(f)
+    def wrapper(*args: Any, **kwargs: Any) -> None:
+        """Wrapper Decorator."""
+        # Nessie context is always as the first argument
+        if len(args) < 1 or not isinstance(args[0], ContextObject):
+            raise RuntimeError("Use this decorator only with Nessie context.")
+
+        ctx: ContextObject = args[0]
+
+        # Check for ref and override with default if is not set
+        if "ref" in kwargs:
```

Comment:
I thought click had decent validation in the library?

Reply:
It does, they have custom validation callback, I tried it but it _seems_ doesn't pass a custom context object (where the nessie client is), thus, I opted for a custom decorator instead